framework/src/modules/chain/bft/finality_manager.js

```diff
@@ -95,13 +98,18 @@ class FinalityManager {
 
 		// Add the header to the list
 		this.headers.add(blockHeader);
-
 		// Update the pre-votes and pre-commits
 		this.updatePreVotesPreCommits(blockHeader);
 
+		// Store current finalizedHeight
+		const currentFinalizedHeight = this.finalizedHeight;
 		// Update the pre-voted confirmed and finalized height
 		this.updatePreVotedAndFinalizedHeight();
 
+		if (currentFinalizedHeight !== this.finalizedHeight) {
+			this.emit(EVENT_NEW_FINALIZED_HEIGHT, blockHeader);
+		}
```

Comment:
@pablitovicente Please move this same logic inside function `updatePreVotedAndFinalizedHeight` This way no matter when and from where the finalized height is changed. The event will be emitted. In one of my PR I am refactoring a bit of code to only call `updatePreVotedAndFinalizedHeight` once during the `recompute`. So that will also fix emitting this event multiple times during recompute process.

Reply:
ok

Reply:
Done and also renamed variables for the code to make a bit more sense.